src/cluster_argument_parsing.rs

```diff
@@ -980,32 +1039,63 @@ pub fn parse_percentage(
 impl GalahClusterer<'_> {
     pub fn cluster(&self) -> Vec<Vec<usize>> {
         let genomes = &self.genome_fasta_paths;
-        let ani = self.ani * 100.;
-        let fastani_min_aligned_threshold = self.min_aligned_fraction;
-        let fastani_fraglen = self.fraglen;
         match self.preclusterer {
-            Preclusterer::Dashing { min_ani, threads } => crate::clusterer::cluster(
-                genomes,
-                &crate::dashing::DashingPreclusterer { min_ani, threads },
-                ani,
-                fastani_min_aligned_threshold,
-                fastani_fraglen,
-            ),
+            Preclusterer::Dashing { min_ani, threads } => match self.clusterer {
```

Comment:
This is getting a bit unwieldy. I tried a `let preclusterer = match`, but it doesn't work since they are different types. Should the Preclusterer/Clusterer enum's be defined using the underlying structs instead of dummy ones?

Reply:
So the issue is that here we have to define the behaviour for every combination of clusterer and preclusterer, right?

I think the answer is yes, well enum or dyn, up to you